src/AzureAppConfigurationImpl.ts

```diff
@@ -13,58 +13,63 @@ import { createCorrelationContextHeader, requestTracingEnabled } from "./request
 import { SettingSelector } from "./types";
 
 export class AzureAppConfigurationImpl extends Map<string, unknown> implements AzureAppConfiguration {
-    private adapters: IKeyValueAdapter[] = [];
+    #adapters: IKeyValueAdapter[] = [];
     /**
      * Trim key prefixes sorted in descending order.
      * Since multiple prefixes could start with the same characters, we need to trim the longest prefix first.
      */
-    private sortedTrimKeyPrefixes: string[] | undefined;
-    private readonly requestTracingEnabled: boolean;
-    private correlationContextHeader: string | undefined;
+    #sortedTrimKeyPrefixes: string[] | undefined;
+    readonly #requestTracingEnabled: boolean;
+    #correlationContextHeader: string | undefined;
+    #client: AppConfigurationClient;
+    #options: AzureAppConfigurationOptions | undefined;
 
     constructor(
-        private client: AppConfigurationClient,
-        private options: AzureAppConfigurationOptions | undefined
+        client: AppConfigurationClient,
+        options: AzureAppConfigurationOptions | undefined
     ) {
         super();
+        this.#client = client;
+        this.#options = options;
+
         // Enable request tracing if not opt-out
-        this.requestTracingEnabled = requestTracingEnabled();
-        if (this.requestTracingEnabled) {
-            this.enableRequestTracing();
+        this.#requestTracingEnabled = requestTracingEnabled();
+        if (this.#requestTracingEnabled) {
+            this.#enableRequestTracing();
         }
 
         if (options?.trimKeyPrefixes) {
-            this.sortedTrimKeyPrefixes = [...options.trimKeyPrefixes].sort((a, b) => b.localeCompare(a));
+            this.#sortedTrimKeyPrefixes = [...options.trimKeyPrefixes].sort((a, b) => b.localeCompare(a));
         }
         // TODO: should add more adapters to process different type of values
         // feature flag, others
-        this.adapters.push(new AzureKeyVaultKeyValueAdapter(options?.keyVaultOptions));
-        this.adapters.push(new JsonKeyValueAdapter());
+        this.#adapters.push(new AzureKeyVaultKeyValueAdapter(options?.keyVaultOptions));
+        this.#adapters.push(new JsonKeyValueAdapter());
     }
 
-    public async load() {
+    async load() {
         const keyValues: [key: string, value: unknown][] = [];
 
         // validate selectors
-        const selectors = getValidSelectors(this.options?.selectors);
+        const selectors = getValidSelectors(this.#options?.selectors);

```

Comment:
I think we'd better make getValidSelector a private static method or it should not be included in this file.

Reply:
I made it top-level method for simplicity. No much difference in JavaScript. But I'm ok to make is a private static method here.

Reply:
No, I decide to leave it as top-level functions at least for now. It would be tedious when I call static private functions as below:
```js
// within class AzureAppConfigurationImpl
const selectors = AzureAppConfigurationImpl.getValidSelectors(this.options?.selectors);
```